gravitee-apim-console-webui/src/management/clusters/details/user-permissions/cluster-user-permissions.component.ts

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, DestroyRef, OnInit, inject } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute } from '@angular/router';
+import { isEmpty, isEqual, uniqueId } from 'lodash';
+import { catchError, combineLatest, EMPTY, filter, forkJoin, Observable, tap } from 'rxjs';
+import { Role } from 'src/entities/role/role';
+import { SearchableUser } from 'src/entities/user/searchableUser';
+import { MemberDataSource } from 'src/permissions/model/member-data-source';
+import { ClusterMemberService } from 'src/services-ngx/cluster-member.service';
+import { RoleService } from 'src/services-ngx/role.service';
+import { SnackBarService } from 'src/services-ngx/snack-bar.service';
+import { UsersService } from 'src/services-ngx/users.service';
+import { GioPermissionService } from 'src/shared/components/gio-permission/gio-permission.service';
+import { GioUsersSelectorComponent, GioUsersSelectorData } from 'src/shared/components/gio-users-selector/gio-users-selector.component';
+import { Member } from '../../../../entities/management-api-v2';
+import { GioTableWrapperFilters } from '../../../../shared/components/gio-table-wrapper/gio-table-wrapper.component';
+
+@Component({
+  selector: 'cluster-user-permissions',
+  templateUrl: './cluster-user-permissions.component.html',
+  styleUrls: ['./cluster-user-permissions.component.scss'],
+  standalone: false,
+})
+export class ClusterUserPermissionsComponent implements OnInit {
+  clusterId: string;
+  dataSource: MemberDataSource[];
```

Comment:
What about : To keep this type only for this component ? 
```suggestion
  dataSource: {
  id: string;
  role: string;
  displayName: string;
  picture: string;
}[];
```